### PR TITLE
Fix issue #129 by rewriting the ARC handler yara rule.

### PR DESCRIPTION
**Pull request by QKaiser**
See #129

### CODE REVIEW

**Comment by QKaiser** [comment]
I added two enhancements to ARC handler that will help with #129. Moved the size to an unsigned integer (should always be positive), and added a name validation function that verify if the file name is unicode encoded.

File: unblob/handlers/archive/arc.py [hunk 1]
```diff
@@ -40,12 +42,19 @@ class ARCHandler(StructHandler):
...
-        long length;	/* true file length */
+        ulong length;	/* true file length */
     };
     """
 
     HEADER_STRUCT = "heads"
 
+    def valid_name(self, name: bytes) -> bool:
+        try:
+            snull(name.decode("utf-8"))
```
Comment (vlaci) [inline unblob/handlers/archive/arc.py:53]: We first want to strip nulls, _then_ convert to string:
```suggestion
            snull(name).decode("utf-8")
```
This'll fix pyright and the test failure too

**Review by vlaci — CHANGES_REQUESTED** [review]

**Comment by vlaci** [comment]
Hm, a file doesn't get extracted now :(

**Comment by vlaci** [comment]
From the docstring of the yara rule:

>             Then a null-byte or unitialized-byte terminated filename string of 13 bytes, the
>             uninitialized byte is always set between 0xf0 and 0xff.

This matches with the output of the test:

> struct heads:
> - archive_marker: 0x1a
> - header_type: 0x2
> - name: b'apple.txt\x00\x00\x00\xfd'
> - size: 0x6
> - date: 0x5379
> - time: 0x6cef
> - crc: 0x3038
> - length: 0x6

This will trip on the `valid_name` check.

**Comment by QKaiser** [comment]
Fixed `valid_name` by moving to this:

```python
# we don't care about the terminating byte (null or unitialized)
snull(name[:-1]).decode("utf-8")
```

**Comment by vlaci** [comment]
Yaay, null bytes are valid unicode code points :D

**Review by vlaci — APPROVED** [review]

**Pull request merged**